services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceTest.kt

```diff
@@ -446,6 +451,157 @@ class HakemusServiceTest {
+        private val tomorrow: LocalDate = today.plusDays(1)
+        private val startDate: ZonedDateTime = ZonedDateTime.parse("2024-08-08T00:00Z")
+        private val beforeStart: LocalDate = LocalDate.parse("2024-08-07")
+        private val endDate: ZonedDateTime = ZonedDateTime.parse("2024-08-12T00:00Z")
+
+        private val id = 16984L
+        private val alluId = 4141
+        private val applicationIdentifier = "JS2400014"
+        private val hanke = HankeFactory.createEntity()
+        private val hakemusData =
+            ApplicationFactory.createExcavationNotificationData(
+                startTime = startDate,
+                endTime = endDate,
+            )
+        private val hakemus =
+            HakemusFactory.createEntity(
+                id = id,
+                alluid = alluId,
+                alluStatus = ApplicationStatus.PENDING,
+                applicationIdentifier = applicationIdentifier,
+                userId = USERNAME,
+                applicationType = ApplicationType.EXCAVATION_NOTIFICATION,
+                hakemusEntityData = hakemusData,
+                hanke = hanke,
+            )
+
+        @ParameterizedTest
+        @ValueSource(
+            strings =
+                [
+                    "2024-08-08", // Start date
+                    "2024-08-09", // After start date
+                    "2024-08-12", // End date
+                    "2024-08-13", // After end date
+                ])
+        fun `sends the date to Allu when the date is between the hakemus start date and today`(
+            date: LocalDate
+        ) {
+            every { hakemusRepository.findOneById(id) } returns hakemus
+            justRun { alluClient.operationalCondition(alluId, date) }
+
+            hakemusService.operationalCondition(hakemus.id, date)
+
+            verifySequence {
+                hakemusRepository.findOneById(id)
+                alluClient.operationalCondition(alluId, date)
+            }
+        }
+
+        @Test
+        fun `throws exception when date is before the start date of the application`() {
+            every { hakemusRepository.findOneById(id) } returns hakemus
+
+            val failure = assertFailure { hakemusService.operationalCondition(id, beforeStart) }
+
+            failure.all {
+                hasClass(OperationalConditionDateException::class)
+                messageContains("Invalid date in operational condition report")
+                messageContains("Date is before the hakemus start date")
+                messageContains("start date: 2024-08-08T00:00")
+                messageContains("date=2024-08-07")
+                messageContains("id=${hakemus.id}")
+            }
+            verifySequence { hakemusRepository.findOneById(id) }
+        }
+
+        @Test
+        fun `throws exception when date is in the future`() {
+            every { hakemusRepository.findOneById(id) } returns
+                hakemus.copy(
```

Comment:
Why adjusting the dates in this test? Isn't it okay to use the same data as in previous test:
`every { hakemusRepository.findOneById(id) } returns hakemus`

Reply:
To check that the important part is being in the future, not being over the hakemus end date.

Reply:
But it can be over the end date. E.g. if the application dates are 8.8.2024 - 12.8.2024, it is perfectly okay to give a date of e.g. 15.8.2024 in the call, right?

Reply:
Yes, that case is tested in the parametrized date test.

Reply:
There the tests do not test a future date. I should have written above 16.8.2024 instead of 15.8.2024. What I mean is that there is no need to change the dates of the application in this test. Or maybe there can be a separate tests cases for application dates that are from past to future and dates that are all in the pasts?

Reply:
Technically, there's no need to change the dates, but I think this way it's clearer for the reader that the exception is thrown because the date is in the future, not because we're past the end date.

Is there a downside to changing the hakemus dates for this test?

Reply:
I guess not. On my first reading the change of dates struck me as something that has to be done in order to make the test pass. Then I realized that the test would pass with the same data as in previous test as well. That's why these questions. It is okay as it is.